pkg/actions/nodegroup/create_test.go

```diff
@@ -0,0 +1,290 @@
+	"github.com/weaveworks/eksctl/pkg/testutils"
+	"github.com/weaveworks/eksctl/pkg/testutils/mockprovider"
+)
+
+type ngEntry struct {
+	version   string
+	opts      nodegroup.CreateOpts
+	mockCalls func(*fakes.FakeKubeProvider, *fakes.FakeNodeGroupInitialiser, *utilFakes.FakeNodegroupFilter)
+	expErr    error
+}
+
+var _ = DescribeTable("Create", func(t ngEntry) {
+	cfg := newClusterConfig()
+	cfg.Metadata.Version = t.version
+
+	p := mockprovider.NewMockProvider()
+	ctl := &eks.ClusterProvider{
+		Provider: p,
+		Status: &eks.ProviderStatus{
+			ClusterInfo: &eks.ClusterInfo{
+				Cluster: testutils.NewFakeCluster("my-cluster", ""),
+			},
+		},
+	}
+	m := nodegroup.New(cfg, ctl, nil)
+
+	k := &fakes.FakeKubeProvider{}
+	m.MockKubeProvider(k)
+
+	init := &fakes.FakeNodeGroupInitialiser{}
+	m.MockNodeGroupService(init)
+
+	ngFilter := utilFakes.FakeNodegroupFilter{}
+	if t.mockCalls != nil {
+		t.mockCalls(k, init, &ngFilter)
+	}
+
+	err := m.Create(t.opts, &ngFilter)
+	if err != nil {
+		Expect(err).To(HaveOccurred())
+		Expect(err).To(MatchError(ContainSubstring(t.expErr.Error())))
+		return
+	}
+
+	Expect(err).NotTo(HaveOccurred())
+},
+	Entry("fails when cluster version is not supported", ngEntry{
+		version: "1.14",
+		mockCalls: func(k *fakes.FakeKubeProvider, init *fakes.FakeNodeGroupInitialiser, f *utilFakes.FakeNodegroupFilter) {
+		},
+		expErr: fmt.Errorf("invalid version, %s is no longer supported, supported values: auto, default, latest, %s\nsee also: https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html", "1.14", strings.Join(api.SupportedVersions(), ", ")),
+	}),
+
+	Entry("fails when it does not support ARM", ngEntry{
+		mockCalls: func(k *fakes.FakeKubeProvider, init *fakes.FakeNodeGroupInitialiser, f *utilFakes.FakeNodegroupFilter) {
+			k.NewRawClientReturns(nil, fmt.Errorf("err"))
+		},
+		expErr: fmt.Errorf("err"),
+	}),
+
+	Entry("fails to load VPC from config", ngEntry{
+		mockCalls: func(k *fakes.FakeKubeProvider, init *fakes.FakeNodeGroupInitialiser, f *utilFakes.FakeNodegroupFilter) {
+			k.NewRawClientReturns(&kubernetes.RawClient{}, nil)
+			k.ServerVersionReturns("1.17", nil)
+			k.LoadClusterIntoSpecFromStackReturns(&manager.StackNotFoundErr{})
+		},
+		expErr: errors.Wrapf(errors.New("VPC configuration required for creating nodegroups on clusters not owned by eksctl: vpc.subnets, vpc.id, vpc.securityGroup"), "loading VPC spec for cluster %q", "my-cluster"),
+	}),
+
+	Entry("fails when cluster does not support managed nodes", ngEntry{
+		mockCalls: func(k *fakes.FakeKubeProvider, init *fakes.FakeNodeGroupInitialiser, f *utilFakes.FakeNodegroupFilter) {
+			k.NewRawClientReturns(&kubernetes.RawClient{}, nil)
+			k.ServerVersionReturns("1.17", nil)
+			k.LoadClusterIntoSpecFromStackReturns(nil)
+			k.SupportsManagedNodesReturns(false, errors.New("err"))
+		},
+		expErr: errors.New("err"),
+	}),
+
+	Entry("fails when NodeGroupService fails to match instances", ngEntry{
```

Comment:
what does this mean? match instances to what?

Reply:
Updated it to make it more descriptive :)